### PR TITLE
Handle null Regex in the config as the empty regex.

### LINKED ISSUE #2149: panic: runtime error: invalid memory address or nil pointer dereference
**What did you do?**
Restarted prometheus

**What did you expect to see?**
Started prometheus
**What did you see instead? Under which circumstances?**

**Environment**

Linux

* System information:
```
Linux 4.4.0-38-generic x86_64
```

* Prometheus version: 
```
prometheus, version 1.3.0 (branch: master, revision: 18254a172b1e981ed593442b2259bd63617d6aca)
  build user:       root@d363f050a0e0
  build date:       20161101-17:06:27
  go version:       go1.7.3
```
* Prometheus configuration file:
```
insert configuration here
```

* Logs:
```
INFO[0726] All requests for rebuilding the label indexes queued. (Actual processing may lag behind.)  source=crashrecovery.go:528
INFO[0726] Checkpointing fingerprint mappings...         source=persistence.go:1372
INFO[0726] Done checkpointing fingerprint mappings in 99.016883ms.  source=persistence.go:1395
WARN[0726] Crash recovery complete.                      source=crashrecovery.go:151
INFO[0726] 98806 series loaded.                          source=storage.go:359
INFO[0726] Starting target manager...                    source=targetmanager.go:76
INFO[0726] Listening on :9090                            source=web.go:240
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x637b3a]

goroutine 275 [running]:
panic(0x1764a00, 0xc4200140a0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
regexp.(*Regexp).FindStringSubmatchIndex(0x0, 0x0, 0x0, 0x7, 0x1, 0x0)
	/usr/local/go/src/regexp/regexp.go:922 +0x7a
github.com/prometheus/prometheus/relabel.relabel(0xc44612d440, 0xc420037a80, 0xc44612d440)
	/go/src/github.com/prometheus/prometheus/relabel/relabel.go:59 +0x6e3
github.com/prometheus/prometheus/relabel.Process(0xc420064e40, 0xc420483620, 0x3, 0x3, 0x6)
	/go/src/github.com/prometheus/prometheus/relabel/relabel.go:35 +0x158
github.com/prometheus/prometheus/retrieval.populateLabels(0xc420064e40, 0xc4204d4480, 0xc481be8c70, 0xc476d043a8, 0x0, 0x1)
	/go/src/github.com/prometheus/prometheus/retrieval/targetmanager.go:461 +0x39e
github.com/prometheus/prometheus/retrieval.targetsFromGroup(0xc420064c60, 0xc4204d4480, 0x0, 0xc4204e6d00, 0x1, 0x0, 0xc495381130)
	/go/src/github.com/prometheus/prometheus/retrieval/targetmanager.go:523 +0x1b1
github.com/prometheus/prometheus/retrieval.(*targetSet).runProviders.func1(0xc452d13510, 0xc498ebbec0, 0xc495381a40, 0xc452d13520, 0xc452d13508, 0x8, 0x2426580, 0xc45741ece0)
	/go/src/github.com/prometheus/prometheus/retrieval/targetmanager.go:298 +0x4d6
created by github.com/prometheus/prometheus/retrieval.(*targetSet).runProviders
	/go/src/github.com/prometheus/prometheus/retrieval/targetmanager.go:330 +0x2dc
```

### PR DESCRIPTION
**Pull request by brian-brazil**
Fixes #2149

### CODE REVIEW

**Comment by beorn7** [comment]
I guess this affects v1.2, too?

With the DNS issues I uncovered yesterday, I'm tempted to cut a 1.2.3...

File: config/config.go [hunk 1]
```diff
@@ -1019,6 +1019,9 @@ func (c *RelabelConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err := checkOverflow(c.XXX, "relabel_config"); err != nil {
 		return err
 	}
+	if (c.Regex == Regexp{}) {
```
Comment (brian-brazil) [inline config/config.go]: Updated

Comment (beorn7) [inline config/config.go]: Another approach: Leave out the `omitempty` in L997. Does that do the trick?

It's kind of weird to first tell the marshaler to ignore an empty regex field to then check for it and create it after all.

**Comment by beorn7** [comment]
Had another idea: Omit the `omitempty`. (Might have drowned as "outdated comment").

**Comment by brian-brazil** [comment]
The omitempty doesn't make a difference, as that only applies at marshalling time.

**Comment by beorn7** [comment]
OK, :+1: 

If you want to save me the cherrypick dance, please rebase onto release-1.2 first, and then merge from there to master.

**Comment by brian-brazil** [comment]
That's on 1.2, not sure how you want that to get to master.

**Comment by beorn7** [comment]
We just merge. I can do that.

Thanks!

**Comment by beorn7** [comment]
Change is now also in master.

**Pull request merged**